load_balancer/test/functional/load_balancerTest.js

```diff
@@ -0,0 +1,449 @@
+    const execSubmissionObj = testData.executedJob;
+    stubConsole();
+    testDetails.requestRunNockObj = requestRunNock(`https://${execSubmissionObj.node_details.hostname}:${execSubmissionObj.node_details.port}`);
+    testDetails.resultsNockObj = resultsNock(false, execSubmissionObj);
+    jobSubmitCheck();
+    submitReqCheck(execSubmissionObj);
+    testDetails.resultsVal = true;
+    testDetails.requestRunVal = true;
+    testAssertion(testDetails, [10], () => {
+      done();
+    });
+  });
+});
+
+describe('Submits a job and receives result correctly', () => {
+  beforeEach(async () => {
+    stubConsole();
+    startMysqlConnection();
+    await sleepFunc(200);
+    startLoadBalancer();
+    restoreConsole();
+    activateNocks();
+  });
+
+  afterEach((done) => {
+    cleanNocks();
+    mysqlConnection.end();
+    loadBalancer.server.close(done);
+  });
+
+  it('for valid case', (done) => {
+    const testDetails = {};
+    const tempJob = testData.submissionsData[0];
+    const execSubmissionObj = testData.executedJob;
+    stubConsole();
+    testDetails.resultsNockObj = resultsNock(false, execSubmissionObj);
+    testDetails.requestRunNockObj = nock(`https://${execSubmissionObj.node_details.hostname}:${execSubmissionObj.node_details.port}`)
+    .persist()
+    .post('/requestRun')
+    .reply(200, (uri, requestBody) => {
+      requestBody.should.deep.equal(tempJob);
+      submitReqCheck(execSubmissionObj);
+      return true;
+    });
+    mockENStatus(`https://${execSubmissionObj.node_details.hostname}:${execSubmissionObj.node_details.port}`, true);
+    mockENStatus(`https://${execSubmissionObj.node_details.hostname}:${execSubmissionObj.node_details.port}`, true);
+    jobSubmitCheck();
+    testDetails.resultsVal = true;
+    testDetails.requestRunVal = true;
+    testAssertion(testDetails, [0, 10], () => {
+      done();
+    });
+  });
+
+  it('with large logs of 30000 lines', (done) => {
+    const testDetails = {};
+    const execSubmissionObj = JSON.parse(JSON.stringify(testData.executedJob));
+    let tempStr = 'Temporary string';
+    let count = 30000;
+    while (count > 0) {
+      tempStr = `Temp \n${tempStr}`;
+      count -= 1;
+    }
+    stubConsole();
+    execSubmissionObj.submission_details.log = tempStr;
+    testDetails.resultsNockObj = resultsNock(false, execSubmissionObj);
+    submitReqCheck(execSubmissionObj);
+    testDetails.resultsVal = true;
+    testAssertion(testDetails, [10], () => {
+      done();
+    });
+  });
+
+  it('when EN sends invalid messages', (done) => {
+    const testDetails = {};
+    const execSubmissionObj = JSON.parse(JSON.stringify(testData.executedJob));
+    stubConsole();
+    execSubmissionObj.submission_details.marks = ['1', '1'];
+    testDetails.resultsNockObj = resultsNock(true, execSubmissionObj);
+    submitReqCheck(execSubmissionObj);
+    testDetails.resultsVal = true;
+    testAssertion(testDetails, [10], () => {
+      done();
+    });
+  });
+});
+
```

Comment:
I don't see any failing tests (tests that we expect to fail). Are they to be added later?

Reply:
Sir, the last test should be a failing test. For now, I've altered the condition so that it passes.
Should I let it fail?